tools/makecorever.py

```diff
@@ -20,23 +20,40 @@
 import argparse
 import os
 import subprocess
+import re
 
 
-def generate(path, platform_path, git_ver="ffffffff", git_desc="unspecified"):
+def generate(path, platform_path, git_ver="ffffffff", platform_version="unspecified"):
     def git(*args):
         cmd = ["git", "-C", platform_path]
         cmd.extend(args)
         proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, universal_newlines=True, stderr=subprocess.DEVNULL)
         return proc.stdout.readlines()[0].strip()
 
+    git_desc = platform_version;
```

Comment:
Following the above, should this have additional param to skip git-describe?
Meaning, this still could still work with a shallow copy or with some broken tagging (like github actions checkout does)

Reply:
Tagging works OK including when used by github-action. Released are based on tagging.
I updated the script per your 3 reviews and added comments.

Reply:
broken == checkout doing `git fetch origin +$commit-sha$:refs/tags/$tag-name$`, which is worked around with fetch-depth:0... which could be avoided

my only problem with that is that version string depends on too much stuff, where we already know it from the tag name anyway; and commit sha comes from rev-parse regardless, shallow copy or not.